common/workunit/workunit.cpp

```diff
@@ -13276,58 +13276,120 @@ bool isValidMemoryValue(const char *memoryUnit)
 }
 
 #ifdef _CONTAINERIZED
-bool executeGraphOnLingeringThor(IConstWorkUnit &workunit, const char *graphName)
+bool executeGraphOnLingeringThor(IConstWorkUnit &workunit, const char *graphName, const char *multiJobLingerQueueName)
 {
     // check if lingering thor instance is up.
 
     /* If code was dependent on reading a workunit in parallel, the whole area of
-     * workunit locking and reading will need revisiting, because at the moment the 
+     * workunit locking and reading will need revisiting, because at the moment the
      * workunit reading code does not lock at all.
-     * 
+     *
+     * Either,
+     * 1) multiJobThorLinger is set, meaning Thor instances will wait for any job's graph.
+     * OR
+     * 2) if not set, the Thor will wait for graphs from same job only, and will be
+     *    notified to quit by the agent, when the last graph is complete.
+     * NB: In both cases, the amount of time the Thor instance stays up waiting for another
+     *     graph is controlled by the Thor setting 'lingerPeriod'.
      */
 
-    /* NB: forcing a reload here to ensure Debug values are recent.
-     * Could be improved by refreshing only the specific area of interest (i.e. Debug/ in this case).
-     * The area of persisting a non-locked connection to workunits should be resivisted..
-     */
-    workunit.forceReload();
-
-    Owned<IStringIterator> iter = &workunit.getDebugValues("thorinstance_*");
-    ForEach(*iter)
+    try
     {
-        /* NB: Thor's set their running endpoint into Debug values of workunit
-         * Check to see if workunit has any Thor's available lingering instances.
-         */
-        SCMStringBuffer thorInstance;
-        iter->str(thorInstance);
-        SCMStringBuffer thorInstanceValue;
-        if (workunit.getDebugValueBool(thorInstance.s, false))
+        if (multiJobLingerQueueName && graphName)
         {
+            // check a Dali queue for idle spare Thor's
+            DBGLOG("Checking multi job lingering queue: %s", multiJobLingerQueueName);
+
+            VStringBuffer xpath("/Status/ThorLinger/%s", multiJobLingerQueueName);
```

Comment:
This code would benefit from a description of the process used to implement a lingering thor - e.g., who is listening to what, and how the next job is kicked off.

Reply:
There's some description at the top of this function, but I'll extend it further.